psi4/driver/qcdb/molecule.py

```diff
@@ -51,14 +56,101 @@ class Molecule(LibmintsMolecule):
+                 fix_com=None,
+                 fix_orientation=None,
+                 fix_symmetry=None,
+
+                 fragment_separators=None,
+                 fragment_charges=None,
+                 fragment_multiplicities=None,
+
+                 molecular_charge=None,
+                 molecular_multiplicity=None,
+
+                 enable_qm=True,
+                 enable_efp=True,
+                 missing_enabled_return_qm='none',
+                 missing_enabled_return_efp='none',
+
+                 missing_enabled_return='error',
+                 tooclose=0.1,
+                 zero_ghost_fragments=False,
+                 nonphysical=False,
+                 mtol=1.e-3,
+                 verbose=1):
         """Initialize Molecule object from LibmintsMolecule"""
-        super(Molecule, self).__init__(psi4molstr=psi4molstr)
+        super(Molecule, self).__init__()
+
+        if molinit is not None or geom is not None:
+            if isinstance(molinit, dict):
+                molrec = molinit
+
+            elif isinstance(molinit, basestring):
+                compound_molrec = molparse.from_string(
+                    molstr=molinit,
+                    dtype=dtype,
+                    name=name,
+                    fix_com=fix_com,
+                    fix_orientation=fix_orientation,
+                    fix_symmetry=fix_symmetry,
+                    return_processed=False,
+                    enable_qm=enable_qm,
+                    enable_efp=enable_efp,
+                    missing_enabled_return_qm=missing_enabled_return_qm,
+                    missing_enabled_return_efp=missing_enabled_return_efp,
+                    verbose=verbose)
+                molrec = compound_molrec['qm']
+
+            elif molinit is None and geom is not None:
+                molrec = molparse.from_arrays(
+                    geom=geom,
+                    elea=elea,
+                    elez=elez,
+                    elem=elem,
+                    mass=mass,
+                    real=real,
+                    elbl=elbl,
+                    name=name,
+                    units=units,
+                    input_units_to_au=input_units_to_au,
+                    fix_com=fix_com,
+                    fix_orientation=fix_orientation,
+                    fix_symmetry=fix_symmetry,
+                    fragment_separators=fragment_separators,
+                    fragment_charges=fragment_charges,
+                    fragment_multiplicities=fragment_multiplicities,
+                    molecular_charge=molecular_charge,
+                    molecular_multiplicity=molecular_multiplicity,
+                    domain='qm',
+                    missing_enabled_return=missing_enabled_return,
+                    tooclose=tooclose,
+                    zero_ghost_fragments=zero_ghost_fragments,
+                    nonphysical=nonphysical,
+                    mtol=mtol,
+                    verbose=verbose)
```

Comment:
When we started out serialize and deserialize this kind of thing, never thought we would end up with structures like this. Makes sense however.